tests/types/test_schema.py

```diff
@@ -0,0 +1,151 @@
+    assert schema == Schema([ColSpec(DataType.boolean)])
+    schema = _infer_schema(np.array([1.1, None], dtype=np.object))
+    assert schema == Schema([ColSpec(DataType.double)])
+
+    # test bytes
+    schema = _infer_schema(np.array([bytes([1])], dtype=np.bytes_))
+
+    assert schema == Schema([ColSpec(DataType.binary)])
+    schema = _infer_schema(np.array([bytearray([1])], dtype=np.bytes_))
+    assert schema == Schema([ColSpec(DataType.binary)])
+
+    # test string
+    schema = _infer_schema(np.array(["a"], dtype=np.str))
+    assert schema == Schema([ColSpec(DataType.string)])
+
+    # test boolean
+    schema = _infer_schema(np.array([True], dtype=np.bool))
+    assert schema == Schema([ColSpec(DataType.boolean)])
+
+    # test ints
+    for t in [np.uint8, np.uint16, np.int8, np.int16, np.int32]:
+        schema = _infer_schema(np.array([1, 2, 3], dtype=t))
+        assert schema == Schema([ColSpec("integer")])
+
+    # test longs
+    for t in [np.uint32, np.int64]:
+        schema = _infer_schema(np.array([1, 2, 3], dtype=t))
+        assert schema == Schema([ColSpec("long")])
+
+    # unsigned long is unsupported
+    with pytest.raises(MlflowException):
+        _infer_schema(np.array([1, 2, 3], dtype=np.uint64))
+
+    # test floats
+    for t in [np.float16, np.float32]:
+        schema = _infer_schema(np.array([1.1, 2.2, 3.3], dtype=t))
+        assert schema == Schema([ColSpec("float")])
+
+    # test doubles
+    schema = _infer_schema(np.array([1.1, 2.2, 3.3], dtype=np.float64))
+    assert schema == Schema([ColSpec("double")])
+
+    # unsupported
+    if hasattr(np, "float128"):
+        with pytest.raises(MlflowException):
+            _infer_schema(np.array([1, 2, 3], dtype=np.float128))
+
+
+def test_that_schema_inference_with_tensors_raises_exception():
+    with pytest.raises(MlflowException):
+        _infer_schema(np.array([[[1, 2, 3]]], dtype=np.int64))
+    with pytest.raises(MlflowException):
+        _infer_schema(pd.DataFrame({"x": [np.array([[1, 2, 3]], dtype=np.int64)]}))
+    with pytest.raises(MlflowException):
+        _infer_schema({"x": np.array([[1, 2, 3]], dtype=np.int64)})
+
+
+def test_spark_schema_inference(pandas_df_with_all_types):
+    try:
+        import pyspark
+        from pyspark.sql.types import _parse_datatype_string, StructField, StructType
+        schema = _infer_schema(pandas_df_with_all_types)
+        assert schema == Schema([ColSpec(x, x) for x in pandas_df_with_all_types.columns])
+        spark_session = pyspark.sql.SparkSession(pyspark.SparkContext.getOrCreate())
+        spark_schema = StructType(
+            [StructField(t.name, _parse_datatype_string(t.name), True)
+             for t in schema.column_types()])
+        sparkdf = spark_session.createDataFrame(pandas_df_with_all_types, schema=spark_schema)
+        schema = _infer_schema(sparkdf)
+        assert schema == Schema([ColSpec(x, x) for x in pandas_df_with_all_types.columns])
+    except ImportError:
```

Comment:
super sketch, why not only run when it's available (e.g., large tests)?

Reply:
oh, good catch, I have no idea why it's not just marked large